Panic on startup CreateApiserverClient

#### Issue details
When the dashboard pod starts it immediately dies with a runtime panic.

##### Environment
```
Dashboard version: 1.4.2
Kubernetes version: 1.4.4
Operating system: CentOS 7
Node.js version:
Go version:
```

##### Steps to reproduce

1.  Launch the dashboard with `kubectl create -f https://rawgit.com/kubernetes/dashboard/master/src/deploy/kubernetes-dashboard.yaml`
2. Open the container logs

##### Observed result

- `kubectl describe pods` returns an infinite loop of _Created container_, _Started container_, _Error syncing pod, skipping: Failed to start container_

- the dashboard container logs contain the following:
```
Using HTTP port: 9090
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x168 pc=0x4ecf12]
goroutine 1 [running]:
panic(0x15316a0, 0xc420014020)
	/usr/lib/google-golang/src/runtime/panic.go:507 +0x1b3
github.com/kubernetes/dashboard/src/app/backend/client.CreateApiserverClient(0x0, 0x0, 0x0, 0x0, 0x1, 0x0, 0xc420068090, 0x2044340, 0xc420437490)
	/usr/local/google/home/rfranken/k8s/dashboard/.tmp/backend/src/github.com/kubernetes/dashboard/src/app/backend/client/apiserverclient.go:48 +0x1d2
main.main()
	/usr/local/google/home/rfranken/k8s/dashboard/.tmp/backend/src/github.com/kubernetes/dashboard/src/app/backend/dashboard.go:58 +0x2f1
```

##### Expected result
The dashboard to start.

##### Comments
It started and worked the first time, then after the first reboot of the node, this behaviour is persistent.
Fresh `kubeadm` installation.
Weavenet

--- a/hooks/pre-commit
+++ b/hooks/pre-commit
@@ -48,7 +48,7 @@ if ${gofmt_available}; then
   echo "${reset}"
   echo -ne "Checking go files formatting... "
   unformatted=$(gofmt -l src/app/backend)
-  if [ ! -z ${unformatted} ]; then
+  if [ ! -z "$unformatted" ]; then
     echo "${red}ERROR!"
     echo "There is an issue with file formatting."
     echo "To format go files run:"

--- a/src/app/backend/client/apiserverclient.go
+++ b/src/app/backend/client/apiserverclient.go
@@ -45,12 +45,12 @@ func CreateApiserverClient(apiserverHost string, kubeConfig string) (*client.Cli
 		&clientcmd.ConfigOverrides{ClusterInfo: clientcmdapi.Cluster{Server: apiserverHost}})
 
 	cfg, err := clientConfig.ClientConfig()
-	cfg.QPS = defaultQPS
-	cfg.Burst = defaultBurst
-
 	if err != nil {
 		return nil, nil, err
 	}
+
+	cfg.QPS = defaultQPS
+	cfg.Burst = defaultBurst
 
 	log.Printf("Creating API server client for %s", cfg.Host)
 

--- a/src/app/backend/resource/common/podmetrics.go
+++ b/src/app/backend/resource/common/podmetrics.go
@@ -138,11 +138,11 @@ func fillPodMetrics(cpuMetrics []heapster.MetricResult, memMetrics []heapster.Me
 			cpuMetricsList := cpuMetrics[iterator].Metrics
 
 			if len(memMetricsList) > 0 {
-				memValue = &memMetricsList[len(memMetricsList) - 1].Value
+				memValue = &memMetricsList[len(memMetricsList)-1].Value
 			}
 
 			if len(cpuMetricsList) > 0 {
-				cpuValue = &cpuMetricsList[len(cpuMetricsList) - 1].Value
+				cpuValue = &cpuMetricsList[len(cpuMetricsList)-1].Value
 			}
 
 			cpuHistory := make([]MetricResult, len(cpuMetricsList))

--- a/src/app/backend/resource/daemonset/daemonsetlist/daemonsetlist.go
+++ b/src/app/backend/resource/daemonset/daemonsetlist/daemonsetlist.go
@@ -24,8 +24,8 @@ import (
 	"k8s.io/kubernetes/pkg/apis/extensions"
 	client "k8s.io/kubernetes/pkg/client/unversioned"
 
-	"github.com/kubernetes/dashboard/src/app/backend/resource/dataselect"
 	"github.com/kubernetes/dashboard/src/app/backend/resource/daemonset"
+	"github.com/kubernetes/dashboard/src/app/backend/resource/dataselect"
 	"github.com/kubernetes/dashboard/src/app/backend/resource/metric"
 )
 

--- a/src/app/backend/resource/dataselect/dataselectquery.go
+++ b/src/app/backend/resource/dataselect/dataselectquery.go
@@ -15,8 +15,8 @@
 package dataselect
 
 import (
-	"github.com/kubernetes/dashboard/src/app/backend/resource/metric"
 	"github.com/kubernetes/dashboard/src/app/backend/resource/common"
+	"github.com/kubernetes/dashboard/src/app/backend/resource/metric"
 )
 
 // Options for GenericDataSelect which takes []GenericDataCell and returns selected data.

--- a/src/app/backend/resource/pod/poddetail.go
+++ b/src/app/backend/resource/pod/poddetail.go
@@ -26,15 +26,14 @@ import (
 
 	"github.com/kubernetes/dashboard/src/app/backend/client"
 	"github.com/kubernetes/dashboard/src/app/backend/resource/common"
+	"github.com/kubernetes/dashboard/src/app/backend/resource/daemonset/daemonsetlist"
 	"github.com/kubernetes/dashboard/src/app/backend/resource/dataselect"
 	"github.com/kubernetes/dashboard/src/app/backend/resource/job/joblist"
+	"github.com/kubernetes/dashboard/src/app/backend/resource/metric"
+	"github.com/kubernetes/dashboard/src/app/backend/resource/petset/petsetlist"
 	"github.com/kubernetes/dashboard/src/app/backend/resource/replicaset/replicasetlist"
 	"github.com/kubernetes/dashboard/src/app/backend/resource/replicationcontroller/replicationcontrollerlist"
-	"github.com/kubernetes/dashboard/src/app/backend/resource/daemonset/daemonsetlist"
-	"github.com/kubernetes/dashboard/src/app/backend/resource/petset/petsetlist"
-	"github.com/kubernetes/dashboard/src/app/backend/resource/metric"
 )
-
 
 // PodDetail is a presentation layer view of Kubernetes PodDetail resource.
 // This means it is PodDetail plus additional augumented data we can get
@@ -76,10 +75,10 @@ type Controller struct {
 	// Singleton list of the Job that controls this Pod, only set if Kind = "Job"
 	JobList *joblist.JobList `json:"joblist,omitempty"`
 
-	ReplicaSetList *replicasetlist.ReplicaSetList `json:"replicasetlist,omitempty"`
+	ReplicaSetList            *replicasetlist.ReplicaSetList                       `json:"replicasetlist,omitempty"`
 	ReplicationControllerList *replicationcontrollerlist.ReplicationControllerList `json:"replicationcontrollerlist,omitempty"`
-	DaemonSetList *daemonsetlist.DaemonSetList `json:"daemonsetlist,omitempty"`
-	PetSetList *petsetlist.PetSetList `json:"petsetlist,omitempty"`
+	DaemonSetList             *daemonsetlist.DaemonSetList                         `json:"daemonsetlist,omitempty"`
+	PetSetList                *petsetlist.PetSetList                               `json:"petsetlist,omitempty"`
 }
 
 // Container represents a docker/rkt/etc. container that lives in a pod.
@@ -138,7 +137,7 @@ func GetPodDetail(client k8sClient.Interface, heapsterClient client.HeapsterClie
 	creatorAnnotation, found := pod.ObjectMeta.Annotations[api.CreatedByAnnotation]
 	if found {
 		creatorRef, err := getPodCreator(client, creatorAnnotation, common.NewSameNamespaceQuery(namespace), heapsterClient)
-		if (err != nil) {
+		if err != nil {
 			return nil, err
 		}
 		controller = *creatorRef
@@ -160,7 +159,7 @@ func GetPodDetail(client k8sClient.Interface, heapsterClient client.HeapsterClie
 
 func getPodCreator(client k8sClient.Interface, creatorAnnotation string, nsQuery *common.NamespaceQuery, heapsterClient client.HeapsterClient) (*Controller, error) {
 	var serializedReference api.SerializedReference
-	err := json.Unmarshal([]byte(creatorAnnotation), &serializedReference);
+	err := json.Unmarshal([]byte(creatorAnnotation), &serializedReference)
 	if err != nil {
 		return nil, err
 	}
@@ -211,7 +210,7 @@ func toJobPodController(client k8sClient.Interface, reference api.ObjectReferenc
 	jobs := []batch.Job{*job}
 	jobList := joblist.CreateJobList(jobs, pods, events, dataselect.StdMetricsDataSelect, &heapsterClient)
 	return &Controller{
-		Kind: "Job",
+		Kind:    "Job",
 		JobList: jobList,
 	}, nil
 }
@@ -224,7 +223,7 @@ func toReplicaSetPodController(client k8sClient.Interface, reference api.ObjectR
 	replicaSets := []extensions.ReplicaSet{*rs}
 	replicaSetList := replicasetlist.CreateReplicaSetList(replicaSets, pods, events, dataselect.StdMetricsDataSelect, &heapsterClient)
 	return &Controller{
-		Kind: "ReplicaSet",
+		Kind:           "ReplicaSet",
 		ReplicaSetList: replicaSetList,
 	}, nil
 }
@@ -251,7 +250,7 @@ func toDaemonSetPodController(client k8sClient.Interface, reference api.ObjectRe
 
 	daemonSetList := daemonsetlist.CreateDaemonSetList(daemonsets, pods, events, dataselect.StdMetricsDataSelect, &heapsterClient)
 	return &Controller{
-		Kind: "DaemonSet",
+		Kind:          "DaemonSet",
 		DaemonSetList: daemonSetList,
 	}, nil
 }
@@ -265,7 +264,7 @@ func toPetSetPodController(client k8sClient.Interface, reference api.ObjectRefer
 
 	petSetList := petsetlist.CreatePetSetList(petsets, pods, events, dataselect.StdMetricsDataSelect, &heapsterClient)
 	return &Controller{
-		Kind: "PetSet",
+		Kind:       "PetSet",
 		PetSetList: petSetList,
 	}, nil
 }

--- a/src/app/backend/resource/replicaset/replicasetcommon.go
+++ b/src/app/backend/resource/replicaset/replicasetcommon.go
@@ -20,7 +20,6 @@ import (
 	"github.com/kubernetes/dashboard/src/app/backend/resource/dataselect"
 	"github.com/kubernetes/dashboard/src/app/backend/resource/metric"
 	"k8s.io/kubernetes/pkg/apis/extensions"
-
 )
 
 // ReplicaSet is a presentation layer view of Kubernetes Replica Set resource. This means

--- a/src/app/backend/resource/replicaset/replicasetlist/replicasetlist.go
+++ b/src/app/backend/resource/replicaset/replicasetlist/replicasetlist.go
@@ -25,8 +25,8 @@ import (
 	"github.com/kubernetes/dashboard/src/app/backend/resource/metric"
 	"github.com/kubernetes/dashboard/src/app/backend/resource/replicaset"
 	"k8s.io/kubernetes/pkg/api"
-	"k8s.io/kubernetes/pkg/apis/extensions"
 	k8serrors "k8s.io/kubernetes/pkg/api/errors"
+	"k8s.io/kubernetes/pkg/apis/extensions"
 	client "k8s.io/kubernetes/pkg/client/unversioned"
 )
 
@@ -35,10 +35,9 @@ type ReplicaSetList struct {
 	ListMeta common.ListMeta `json:"listMeta"`
 
 	// Unordered list of Replica Sets.
-	ReplicaSets       []replicaset.ReplicaSet    `json:"replicaSets"`
-	CumulativeMetrics []metric.Metric `json:"cumulativeMetrics"`
+	ReplicaSets       []replicaset.ReplicaSet `json:"replicaSets"`
+	CumulativeMetrics []metric.Metric         `json:"cumulativeMetrics"`
 }
-
 
 // GetReplicaSetList returns a list of all Replica Sets in the cluster.
 func GetReplicaSetList(client client.Interface, nsQuery *common.NamespaceQuery,

--- a/src/app/backend/resource/replicationcontroller/replicationcontrollerlist/replicationcontrollerlist.go
+++ b/src/app/backend/resource/replicationcontroller/replicationcontrollerlist/replicationcontrollerlist.go
@@ -33,9 +33,8 @@ type ReplicationControllerList struct {
 
 	// Unordered list of Replication Controllers.
 	ReplicationControllers []replicationcontroller.ReplicationController `json:"replicationControllers"`
-	CumulativeMetrics      []metric.Metric         `json:"cumulativeMetrics"`
+	CumulativeMetrics      []metric.Metric                               `json:"cumulativeMetrics"`
 }
-
 
 // GetReplicationControllerList returns a list of all Replication Controllers in the cluster.
 func GetReplicationControllerList(client *client.Client, nsQuery *common.NamespaceQuery,

